src/tools/simeng/main.cc

```diff
@@ -87,6 +89,11 @@ int main(int argc, char** argv) {
     mode = SimulationMode::OutOfOrder;
   }
 
+#ifdef GDB_ENABLED
+  std::cout << "GDB Enabled\n";
+  mode = SimulationMode::Emulation;
+#endif
```

Comment:
This `#ifdef GDB_ENABLED`  clause could encapsulate the original SimulationMode setting logic as currently it could be doing this logic redundantly. i.e. add an else clause to your `#ifdef` which surrounds the setting of `SimulationMode` from the config file.

Reply:
thanks - added in a later commit